src/InteractiveWindow/VisualStudio/VsInteractiveWindowCommandFilter.cs

```diff
@@ -330,7 +330,19 @@ private int PreLanguageCommandFilterExec(ref Guid pguidCmdGroup, uint nCmdID, ui
                 switch ((CommandIds)nCmdID)
                 {
                     case CommandIds.AbortExecution: _window.Evaluator.AbortExecution(); return VSConstants.S_OK;
-                    case CommandIds.Reset: _window.Operations.ResetAsync(); return VSConstants.S_OK;
+                    case CommandIds.Reset:
+                            try
+                            {
+                                ((IInteractiveWindowOperations2)(_window.Operations)).ResetAsync(isFromSubmit: false); return VSConstants.S_OK;
+                            }
+                            catch (InvalidCastException)
+                            {
+                                // in cases where the cast to IInteractiveWindowOperations2 fails when the instance passed in
+                                // does not implement IInteractiveWindowOperations2 then we fall back to IInteractiveWindowOperations
+                                // ResetAsync method
+                                _window.Operations.ResetAsync(); return VSConstants.S_OK;
+                            }
```

Comment:
Perhaps add an extension method ResetAsync(this IInteractiveWindowOperations ops, bool initialize, bool isFromSubmit) that uses IInteractiveWindowOperations2 if available. Then that method is available to all callers.